python/ray/serve/tests/test_http_cancellation.py

```diff
@@ -96,5 +99,103 @@ async def __call__(self, *args):
         assert h.remote().result() == i
 
 
+async def test_request_cancelled_error_on_http_client_disconnect_during_execution(
+    serve_instance,
+):
+    """Test the exception thrown for executing request on http client disconnect"""
+    collector = Collector.remote()
+    child_signal = SignalActor.remote()
+
+    @serve.deployment(max_ongoing_requests=1)
+    class Child:
+        async def __call__(self):
+            try:
+                await child_signal.wait.remote()
+            except asyncio.CancelledError:
+                await collector.add.remote("Child_CancelledError")
+                raise
+
+    @serve.deployment
+    class Parent:
+        def __init__(self, child):
+            self.child = child
+
+        async def __call__(self):
+            try:
+                await self.child.remote()
+            except asyncio.CancelledError:
+                await collector.add.remote("Parent_CancelledError")
+                raise
+            except RequestCancelledError:
+                await collector.add.remote("Parent_RequestCancelledError")
+                raise
+
+    serve.run(Parent.bind(Child.bind()))
+
+    # Make a request with short timeout that will cause disconnection
+    try:
+        await httpx.AsyncClient(timeout=0.1).get("http://localhost:8000/")
+    except httpx.ReadTimeout:
+        pass
+
+    await asyncio.sleep(0.5)
```

Comment:
Why is an additional `asyncio.sleep(0.5)` needed here? generally it's best to avoid adding sleeps in tests.